### PR TITLE
ZIOS-11483: Add logic to create legal hold client from request

### PR DESCRIPTION
**Pull request by alexaubry**
### Issues

We want to create a legal hold client when the user accepts a legal hold.

### Solutions

Create an entity when the user accepts the request, and establish a session from the data in the request.

### CODE REVIEW

File: Source/Model/User/ZMUser+LegalHoldRequest.swift [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 import Foundation
 
+private let log = ZMSLog(tag: "legal-hold")
```
Comment (typfel) [inline Source/Model/User/ZMUser+LegalHoldRequest.swift]: How about using the already existing `UserClient` tag?

File: Tests/Source/Model/User/ZMUserLegalHoldTests.swift [hunk 2]
```diff
@@ -29,36 +29,34 @@ class ZMUserLegalHoldTests: ModelObjectsTests {
         XCTAssertEqual(selfUser.legalHoldStatus, .disabled)
     }
 
-    func testThatLegalHoldStatusIsPending_AfterReceivingRequest() {
+    func testThatLegalHoldStatusIsEnabled_AfterReceivingRequest() {
```
Comment (typfel) [inline Tests/Source/Model/User/ZMUserLegalHoldTests.swift]: Why did you rename this test method? Previous title seems to match what we are checking now.

Comment (alexaubry) [reply Tests/Source/Model/User/ZMUserLegalHoldTests.swift]: Right, I got confused, will revert.

**Comment by alexaubry** [comment]
@zenkins release major
